src/containers/pages/InterventionPlan/IRS/plan/index.tsx

```diff
@@ -1867,37 +1941,50 @@ class IrsPlan extends React.Component<
     country: JurisdictionsByCountry,
     filteredJurisdictions: Jurisdiction[]
   ) {
-    const { point, target: Map } = e;
-    const features = Map.queryRenderedFeatures(point);
-    const { ADMN0_EN, bounds, jurisdictionId, tilesets } = country;
+    const { geographicLevel, clickedId, point, target: Map } = e;
+    const features = point ? Map.queryRenderedFeatures(point) : [];
+    const { ADMN0_EN, jurisdictionId, tilesets } = country;
+    const bounds = e.bounds || country.bounds;
     if (!features.length && tilesets && bounds) {
       if (tilesets.length) {
-        let t = 1;
+        let t = geographicLevel || 1;
         for (t; t < tilesets.length; t += 1) {
           const layerId = `${ADMN0_EN}-admin-${t}-fill`;
           if (Map.getLayer(layerId)) {
             // Reset layers to default visibility
-            if (t === 1) {
+            if (t === 1 && geographicLevel !== t) {
               const isLayerVisible = Map.getLayoutProperty(layerId, 'visibility') === 'visible';
               if (!isLayerVisible) {
-                store.dispatch(Actions.toggleLayer(MAP_ID, layerId));
+                Map.setLayoutProperty(layerId, 'visibility', 'visible');
               }
             } else {
               const layerFilter = Map.getFilter(layerId);
               if (layerFilter) {
-                layerFilter[2] = '';
+                layerFilter[2] = t === geographicLevel + 1 ? clickedId : '';
```

Comment:
Could we make it a little more clearer why we are taking the third member of the array?  Is this always going to be the case, should it be configurable?